src/lib.rs

```diff
@@ -0,0 +1,160 @@
+use lazy_static::lazy_static;
+
+use guest::prelude::*;
+use kubewarden_policy_sdk::wapc_guest as guest;
+
+use k8s_openapi::api::core::v1 as apicore;
+
+extern crate kubewarden_policy_sdk as kubewarden;
+use kubewarden::{logging, protocol_version_guest, request::ValidationRequest, validate_settings};
+
+mod settings;
+use settings::Settings;
+
+use slog::{o, warn, Logger};
+
+lazy_static! {
+    static ref LOG_DRAIN: Logger = Logger::root(
+        logging::KubewardenDrain::new(),
+        o!("policy" => "sample-policy")
+    );
+}
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+fn mutate_request(pvc: apicore::PersistentVolumeClaim, settings: &Settings) -> CallResult {
+    warn!(
+        LOG_DRAIN,
+        "storage class \"{}\" is not allowed, using fallback storage class",
+        settings.fallback_storage_class.clone().unwrap_or_default()
+    );
+    let mutated_pvc = serde_json::to_value(apicore::PersistentVolumeClaim {
+        spec: Some(apicore::PersistentVolumeClaimSpec {
+            storage_class_name: settings.fallback_storage_class.clone(),
+            ..pvc.spec.unwrap_or_default()
+        }),
+        ..pvc
+    })?;
+    kubewarden::mutate_request(mutated_pvc)
+}
+
+fn validate(payload: &[u8]) -> CallResult {
+    let validation_request: ValidationRequest<Settings> = ValidationRequest::new(payload)?;
+
+    // TODO: you can unmarshal any Kubernetes API type you are interested in
+    match serde_json::from_value::<apicore::PersistentVolumeClaim>(
+        validation_request.request.object,
+    ) {
```

Comment:
I prefer to use guards to have a leaner code.

Consider changing the code to something like that:

```rust 
let pvc = match serde_json::from_value<apicore::PersistentVolumeClaim>(obj) {
   Ok(pvc) => pvc,
   _ => {
      // not a PVC object, moving forward
      return accept_request();
   }
};

let (deniedStorageClassFound, storageClassName) = uses_denied_storage_class();
// keep going
```